File: src/problem053/solution053.py
import math

def choose(n, k):
    return math.factorial(n) / (math.factorial(k) * math.factorial(n - k))

def main():
    print(len([1 for n in xrange(1, 101) for k in xrange(0, n + 1) if choose(n, k) >= 1000000]))

if __name__ == '__main__':
    main()

File: src/problem053/solution053.exs
defmodule Math do
  def factorial(n) do
    case n do
      0 -> 1
      m -> m * factorial(m - 1)
    end
  end
  def combinations(n, m) do
    factorial(n) / (factorial(m) * factorial(n - m))
  end
end

IO.puts length(for n <- 1..100, m <- 1..n, Math.combinations(n, m) > 1000000, do: 1)
